### PR TITLE
[TECH]: add support for oauth providers

### PR DESCRIPTION
**Pull request by eliobricenov**
How to test: https://app.qase.io/case/HC-144

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Review by red-game-dev — APPROVED** [review]
LGTM 🍾

File: src/common/constants.ts [hunk 1]
```diff
@@ -8,4 +8,6 @@ export const METAMASK_SNAPS_URL = 'https://snaps.metamask.io/'
 export const METAMASK_SNAPS_EXTERNAL_URL = 'https://metamask.io/snaps/'
 export const METAMASK_PORTFOLIO_URL = 'https://portfolio.metamask.io/'
 
-export const DEV_PORTAL_URL = 'https://developers.hyperplay.xyz/'
+// will be replaced with production url once we merge the changes on the dev portal
+export const DEV_PORTAL_URL =
+  'https://hyperplay-dev-git-feature-unified-auth-ui-valist.vercel.app/'
```
Comment (BrettCleary) [inline src/common/constants.ts:13]: just making one more note here to update this one before merging :)

**Review by BrettCleary — APPROVED** [review]
LGTM. Tested account creation with Discord and account linking with Twitch. All worked great!

**Review by flavioislima — APPROVED** [review]
good stuff

File: src/backend/protocol.ts [hunk 2]
```diff
@@ -93,6 +91,8 @@ export function parseUrl(url: string): [Command, Runner?, string?, string?] {
 
   const urlObject = new URL(url)
 
+  // TODO: make this function more dynamic, ideally only get the command and
+  // handle the params in a different function
```
[the remark this answers is not in the document]
Comment (eliobricenov) [reply src/backend/protocol.ts:95]: niiiiiiiiceeee thanks

**Comment by nyghtstalker** [comment]
1. Flickering and change in font on the Manage Accounts pop-up when opened. 

https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/82125535/2858f858-abc0-4b64-9b2f-713e0c2b14cb

2. Another pop-up triggers when opening Manage Accounts. This may have started after I unlocked my wallet.

https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/assets/82125535/aa0ce7a0-14cc-49a0-9839-60e77eebda81

**Comment by eliobricenov** [comment]
> 1. Flickering and change in font on the Manage Accounts pop-up when opened.
> 
>  ManageAccountsFlickering.mp4 
> 2. Another pop-up triggers when opening Manage Accounts. This may have started after I unlocked my wallet.
> 
>  PopupChangedWhenOpened.mp4

this is fine, I will handle this polishing in a separate PR

**Pull request merged**